Add auto extraction of FireHol lists. Closes #548

opbot-xd opened the pull request:
# Description

Implements auto-extraction of FireHol lists to enhance IOC classification and improve threat intelligence.

This feature enables GreedyBear to:
- **Track unique IPs**: Identify IOCs that aren't widespread mass scanners
- **Classify threat sources**: Categorize IPs based on FireHol list categories (brute force, spam, etc.)
- **Enrich feeds**: Provide users with more context about the nature of threats

## Changes

- Created [FireHolList](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/models.py:32:0-40:9) model to store IPs from FireHol blocklists
- Added `firehol_categories` field to [IOC](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/models.py:43:0-75:24) model for classification metadata
- Implemented [FireHolCron](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/cronjobs/firehol.py:5:0-49:16) job to fetch 4 key lists weekly:
  - `blocklist_de`: IP addresses involved in attacks
  - `greensnow`: Known scanning IPs
  - `bruteforceblocker`: Brute force attack sources
  - `dshield`: DShield top attackers (CIDR blocks)
- Added [extract_firehol_lists](cci:1://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/tasks.py:73:0-77:27) Celery task with weekly scheduling (Sundays 4:15 AM)
- Registered [FireHolList](cci:2://file:///home/opbotxd/Desktop/vscode/dev/gsoc/honeynet/GreedyBear/greedybear/models.py:32:0-40:9) in Django admin for data management
- Exposed `firehol_categories` in Feeds API responses
- Added comprehensive unit tests with 100% coverage

## Related issues
Closes #548

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/GreedyBear/Contribute/) to this project.
- [x] The pull request is for the branch `develop`.
- [ ] I have added documentation of the new features.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors.
- [x] I have added tests for the feature/bug I solved. All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated.
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

mlodic commented on greedybear/cronjobs/firehol.py:
```diff
@@ -0,0 +1,50 @@
+            self.log.info(f"Processing {source} from {url}")
+            try:
+                response = requests.get(url, timeout=60)
+                if response.status_code != 200:
```
a response.raise_for_status() is enough and more comprehensive that this check.

mlodic commented on greedybear/cronjobs/firehol.py:
```diff
@@ -0,0 +1,50 @@
+            except Exception as e:
+                self.log.exception(f"Unexpected error processing {source}: {e}")
+
+    def _update_ioc(self, ip_address, source):
```
considering this, we should also add an additional routine at the end of this cron that deletes the old FireHol items because they are not useful anymore in the database. That database could become really big so it is important to keep it clean

mlodic reviewed (changes requested):
great, we are almost done!

mlodic reviewed (changes requested):
@opbot-xd great job, there's one last thing: another PR has been merged before this so we need to align the migration numbers. See: https://github.com/intelowlproject/GreedyBear/tree/develop/greedybear/migrations, there's already a n.23 migration so your new migrations must be 24 and 25 and should depend on that new one.
To do that ,please pull changed from develop and adjust the migrations. Then we are finally done.

Please @regulartim don't merge other PRs with new migrations to avoid additional conflicts in the meanwhile

opbot-xd commented:
Done! I've synced my fork with the upstream develop branch and fixed the migration conflicts. Since PR #643 was merged with migration 0023_rename_massscanners_massscanner_and_more, I've renumbered our migrations accordingly: the FireHol categories migration is now 0024_ioc_firehol_categories_alter_statistics_view_and_more (depending on the new 0023), and the merge migration is now 0025_merge_20251223_2100. I've also updated all the dependencies to point to the correct parent migrations. Migrations have been tested and all 277 tests are passing.
